test/app/model/ChessEngine.spec.ts

```diff
@@ -142,4 +143,118 @@ describe('Chess Engine', () => {
       ),
     ).toThrowError('There is no piece on this square.');
   });
+
+  describe('If player wants to move piece that check if it causes checking of his king.', () => {
+    const errorMessage = `The player cannot move piece which causes check of his king.`;
```

Comment:
The player cannot move piece which causes checking of his king

Reply:
:)))))) ok ok-